src/app/components/RouterTabs/index.tsx

```diff
@@ -18,13 +18,21 @@ function getPathname(tab: { to: string }) {
 
 export function RouterTabs<Context>({ tabs, context }: RouterTabsProps<Context>) {
   const { pathname } = useLocation()
-  const currentTab = tabs.find(tab => getPathname(tab) === pathname)
+  let targetTab = tabs.find(tab => getPathname(tab) === pathname)
+  const matches = useMatches()
+
+  if (!targetTab) {
+    /// the last index is the current route, -2 is the first parent in route hierarchy
+    const parentPathname = matches?.at(-2)?.pathname
```

Comment:
Haven't run this, but hardcoding this to 2 level deep seems like a wrong approach. Can we take the first child of `matches` as the top parent? To support "true" hierarchy?

Reply:
oh yeah I just need to handle a direct child routes - I've updated commit messages to be more specific. "The first child of `matches`" is root path '/' I am not sure how it can help. To handle any number of nested child routes we would need to do sth like: based on a current route, find an index in `matches` based on the closest `to` value in `tabs` array.